lib/developer_portal/app/controllers/developer_portal/admin/applications/alerts_controller.rb

```diff
@@ -61,15 +61,18 @@ def authorize_alerts
   end
 
   def find_application
-    @cinstance = current_account.bought_cinstances.find(params[:application_id])
+    @cinstance = current_account.bought_cinstances.find(params.require(:application_id))
   end
 
   def collection
-    current_account.alerts.not_deleted.by_application(@cinstance).sorted.order_by(params[:sort], params[:direction])
+    current_account.alerts.not_deleted.by_application(@cinstance).sorted.order_by(local_params[:sort], local_params[:direction])
   end
 
   def resource
-    collection.find(params[:id])
+    collection.find(params.require(:id))
   end
 
+  def local_params
```

Comment:
```suggestion
  private

  def local_params
```

I think it's a good practice to make this private